proxy/ProxySession.h

```diff
@@ -141,6 +141,8 @@ class ProxySession : public VConnection
   TSHttpHookID get_hookid() const;
   bool has_hooks() const;
 
+  virtual bool is_tls() const;
```

Comment:
What's the criteria? Is QUIC (or H3) TLS?

Reply:
I don't know.  What I am really checking is whether there is a SNI possible.  As you point out in the next comment, would be more appropriate to check for SNI.

Reply:
https://github.com/apache/trafficserver/pull/6278 includes a strategy for reusing SSL/TLS logic for H1.1+, H2, H3 sessions:  https://github.com/ywkaras/trafficserver/blob/39c1f233f7a96e0822c0e2c79aec06ab4fef5d59/proxy/ProxySession.h#L149 .  ( @a-canary would have to sign off on this strategy as well.)  We may want to consider using multiple inheritance with virtual inheritance ( aka "diamond" inheritance).